src/pages/trader-tools/pip-calculator/_pip-calculator.tsx

```diff
@@ -78,11 +79,66 @@ const PipCalculator = () => {
     const data = useStaticQuery(query)
 
     const [tab, setTab] = useState('Synthetic')
+    const [activeSymbols, setActiveSymbols] = useState([])
+    const [syntheticSymbolNames, setSyntheticSymbolNames] = useState({})
+    const [financialSymbolNames, setFinancialSymbolNames] = useState({})
 
     const onTabClick = (t) => {
         setTab(t)
     }
 
+    const deriv_api = useDerivApi()
+
+    useEffect(() => {
+        const { send } = deriv_api
+        send({ active_symbols: 'full' }, (response: DerivAPIResponse) => {
+            if (!response.error && response.active_symbols.length > 0) {
+                const data = response.active_symbols
+                setActiveSymbols(data)
+            }
+        })
+    }, [])
+
+    useEffect(() => {
+        const tempSyntheticSymbolNames = []
+
+        if (activeSymbols.length < 1) {
+            return
+        }
+        const data = activeSymbols.filter((activeSymbol) => {
+            return activeSymbol.market === 'synthetic_index'
+        })
+        tempSyntheticSymbolNames.push(data)
+        setSyntheticSymbolNames(tempSyntheticSymbolNames)
+    }, [activeSymbols])
+
+    useEffect(() => {
+        const tempFinancialSymbolNames = []
+
+        if (activeSymbols.length < 1) {
+            return
+        }
+        const data = activeSymbols.filter((activeSymbol) => {
```

Comment:
Hi @konstantinosG-deriv , you can shorten this whole logic. Put all the markets in an array, if the array includes  `activeSymbol.market` you push the data to the `tempFinancialSymbolNames` array